src/transformers/models/mamba2/modeling_mamba2.py

```diff
@@ -975,10 +1036,6 @@ def prepare_inputs_for_generation(
     ):
         # Overwitten -- uses `cache_params` as opposed to `past_key_values`
 
-        if inputs_embeds is not None:
-            past_len = inputs_embeds.shape[1] + input_ids.shape[1]
-        else:
-            past_len = input_ids.shape[1]
```

Comment:
Wasn't really necessary as we only need `conv_kernel` for cache pos as explained below. Same as in Mamba1.